bazel/external/v8.patch

```diff
@@ -27,6 +28,25 @@ index e957c0fad3..063627b72b 100644
                  # Use GNU dialect, because GCC doesn't allow using
                  # ##__VA_ARGS__ when in standards-conforming mode.
                  "-std=gnu++17",
+@@ -151,6 +152,18 @@ def _default_args():
+                 "-fno-integrated-as",
+             ],
+             "//conditions:default": [],
++        }) + select({
++            "@v8//bazel/config:is_macos": [
++                # The clang available on macOS catalina has a warning that isn't clean on v8 code.
++                "-Wno-range-loop-analysis",
++
++                # To supress warning on deprecated declaration on v8 code. For example:
++                # external/v8/src/base/platform/platform-darwin.cc:56:22: 'getsectdatafromheader_64'
++                # is deprecated: first deprecated in macOS 13.0.
++                # https://bugs.chromium.org/p/v8/issues/detail?id=13428.
++                "-Wno-deprecated-declarations",
++            ],
++            "//conditions:default": [],
+         }),
+         includes = ["include"],
+         linkopts = select({
```

Comment:
Nitpicking, but this should be separate PR, since it's unrelated to rules_python / rules_fuzzing.

Reply:
I'm confused because I first ran into this getsectdatafromheader_64 error during this PR's CI. I think it's related to this PR since v8 definitely has Python dependencies.

Reply:
Also I believe that @mpwarres v8 update will fix it. Let's move ahead and we'll plan to roll back this patch soon.

Reply:
> I'm confused because I first ran into this getsectdatafromheader_64 error during this PR's CI. I think it's related to this PR since v8 definitely has Python dependencies.

It's not related. You saw it here first, because all previous builds were broken because of an outdated `rules_python`, and now that this issue is resolved, you've ran into the next issue with compiler warnings, but that's unrelated to the Python changes.

I'm nitpicking about this stuff, because over time, it's really hard to debug, track and/or reason about old issues when fixes are smuggled as part of unrelated PRs, and splitting PRs like this is virtually free (sans wait time on the CI to pass).